corehq/apps/locations/tests/test_location_import.py

```diff
@@ -49,6 +55,70 @@ def test_import_with_existing_parent_by_site_code(self):
         new_loc = Location.get(result['id'])
         self.assertEqual(new_loc.parent_id, self.test_state._id)
 
+    def test_import_with_location_id(self):
+        """
+        When updating with location_id and site_code, location_id should
+        take precedence in uniqueness over site_code.
+        """
+        data = {
+            'name': 'importedloc',
+            'site_code': 'man',
+        }
+
+        result = import_location(self.domain.name, 'state', data)
+
+        new_loc = Location.get(result['id'])
+        data = {
+            'location_id': new_loc._id,
+            'name': 'updatedLoc',
+            'site_code': 'woman',
+        }
+
+        result = import_location(self.domain.name, 'state', data)
+        updated_loc = Location.get(result['id'])
+        self.assertEqual(updated_loc._id, new_loc._id)
+        self.assertEqual(updated_loc.name, 'updatedLoc')
+        self.assertEqual(updated_loc.site_code, 'woman')
+
+    def test_import_with_missing_location_id(self):
```

Comment:
nit: "bad" might be a better descriptor than "missing", since "missing" sounds like it could mean that no location ID is set, which is perfectly valid (although it'll have different behavior)